client/utils/utils.go

```diff
@@ -62,6 +62,22 @@ func CompleteAndBroadcastTxCLI(txBldr authtxb.TxBuilder, cliCtx context.CLIConte
 		return nil
 	}
 
+	if !cliCtx.SkipConfirm {
+		stdSignMsg, err := txBldr.BuildSignMsg(msgs)
+		if err != nil {
+			return err
+		}
+
+		fmt.Fprintf(os.Stderr, "%s\n\n", cliCtx.Codec.MustMarshalJSON(stdSignMsg))
+
+		buf := client.BufferStdin()
+		ok, err := client.GetConfirmation("confirm transaction before signing and broadcasting", buf)
```

Comment:
It looks like this signs the transaction, then asks for confirmation of the signing? Seems a bit off?

Reply:
I don't follow? This asks to confirm the tx _before_ signing and broadcasting which happen on subsequent lines.

Reply:
Ah. Thank you for getting me straight here. I was confused by the `txBldr.BuildSignMsg` call (looked at function, understand now)